packages/web3/src/reads/app/index.ts

```diff
@@ -0,0 +1,96 @@
+import { Address } from 'viem'
+import { CacheManager } from '../cache/cache-manager'
+import { ChainId } from '../clients/get-chain'
+import { createReadClient } from '../clients/readClient'
+import { makeSpaceFactoryReads, SpaceFactoryReads } from '../contracts/space-factory'
+import { bannedWallets, walletAddressIsBanned } from '../aggregators/banned-wallets'
+import {
+    getLinkedWallets,
+    getLinkedWalletsWithDelegations,
+    getMainnetDelegationsForLinkedWallets,
+} from '../aggregators/get-wallets'
+import { XchainConfig } from '../../space/entitlements/entitlement'
+
+export type ReadApp = ReturnType<typeof createReadApp>
+
+export const createReadApp = (args: {
+    chainId: ChainId
+    url: string
+    spaceFactoryAddress: Address
+}): {
+    wallets: SpaceFactoryReads['walletLink'] & {
+        bannedWallets: (args: { spaceId: string }) => Promise<Address[]>
+        walletAddressIsBanned: (
+            args: {
+                spaceId: string
+                walletAddress: Address
+            },
+            opts?: { skipCache?: boolean },
+        ) => Promise<boolean>
+        getLinkedWallets: (args: { walletAddress: Address }) => Promise<Address[]>
+        getLinkedWalletsWithDelegations: (args: {
+            walletAddress: Address
+            xchainConfig: XchainConfig
+        }) => Promise<Address[]>
+        getMainnetDelegationsForLinkedWallets: (args: {
+            linkedWallets: Address[]
+            xchainConfig: XchainConfig
+        }) => Promise<Set<string>>
+    }
+    pricingModules: SpaceFactoryReads['pricingModules']
+    platformRequirements: SpaceFactoryReads['platformRequirements']
+    cacheManager: CacheManager
+    readClient: ReturnType<typeof createReadClient>
+} => {
+    const cacheManager = new CacheManager()
+
+    const readClient = createReadClient({
```

Comment:
nit: let user pass viem client to `createReadApp` instead.
it will grant more options / reusability